clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderFollowerStoreIngestionTask.java

```diff
@@ -565,6 +566,10 @@ protected void checkLongRunningTaskState() throws InterruptedException {
                 consumerTaskId,
                 currentLeaderTopic,
                 partition);
+            if (isDataRecovery && partitionConsumptionState.isBatchOnly() && !versionTopic.equals(currentLeaderTopic)) {
+              partitionConsumptionState.getOffsetRecord().setLeaderTopic(versionTopic);
```

Comment:
Thanks Shuhui for making this change. Just for my knowledge, for changes on the venice-server side, it looks like we can pretty much rely on the existing implementation of the data recovery except there are places that we don't update the `LeaderTopic` properly, is my understanding right?

Reply:
Yes, existing implementation does not support replicating data from a version topic that has a different version number. Changes in this commit add such support by updating `LeaderTopic`.